README.md

```diff
@@ -1 +1,364 @@
+
+# LS AAI registration
+In order to gain access to the LS AAI and enable the users to login with their home organization credentials, the Auth service needs to be registered with LS AAI. The registration process is described [here](https://docs.google.com/document/d/17pNXM_psYOP5rWF302ObAJACsfYnEWhjvxAHzcjvfIE/edit)
+
+The details for the LS AAI configuration are described [here](https://lifescience-ri.eu/ls-login.html). For more information contact Dominic František Bučík on GDI slack.
+
+## Create certificates for all these services
+The user facing services are running with TLS encryption. To create the certificates, we used Certbot. The certificates are stored in a Docker volume, which is mounted in all the containers.
+
+Clone the `storage-and-interfaces` repository using the following command:
+```bash
+git clone https://github.com/GenomicDataInfrastructure/starter-kit-storage-and-interfaces.git
+```
+
+Generate a certificate for all external facing services with a command similar to this (be sure to change the email and the domains):
+```bash
+certbot certonly --standalone \
+        --noninteractive \
+        --agree-tos \ 
+        --preferred-challenges http \
+        --email sda-ops@nbis.se \
+        -d htsget.gdi.nbis.se \
+        -d login.gdi.nbis.se \
+        -d download.gdi.nbis.se \
+        -d beacon.gdi.nbis.se \
+        -d inbox.gdi.nbis.se \
+        -d rems.gdi.nbis.se --expand
+```
+
+To renew the certificates and copy them to the Docker volume, run the following command or add it to the Certbot cron job, for example:
+```none
+1 2	1 * * root /usr/bin/certbot renew --deploy-hook /home/ubuntu/starter-kit-storage-and-interfaces/scripts/cert-deploy --quiet --no-self-upgrade
+```
+
+## Deploying storage and interfaces
+Now that the storage and interfaces repository is cloned, start by editing the `config/config.yml` file (a sample can be found under `storage-and-interfaces/config` in this repository) and add the certificates you generated to the `app` entry (which is the download service), like:
+```yaml
+app: # this is for download
+  host: "0.0.0.0"
+  servercert: /shared/cert/fullchain.pem
+  serverkey: /shared/cert/privkey.pem
+  port: "8443"
+```
+
+And to the `s3inbox` entry in the `docker-compose.yml` file (a sample can be found under `storage-and-interfaces` in this repository) like:
+```yaml
+s3inbox:
+  environment:
+    - SERVER_CERT=/shared/cert/fullchain.pem
+    - SERVER_KEY=/shared/cert/privkey.pem
+```
+
+### Deploying auth
+Auth is a service in the storage-and-interfaces Docker compose file that enables the users to extract a token and an s3config file in order to download and upload data, respectively. 
+In the `docker-compose.yml` file, apart from the credentials of the LS AAI account (registered in earlier steps), you should also set the redirect URL, to something like:
+```yaml
+- ELIXIR_REDIRECTURL=https://login.gdi.nbis.se/elixir/login
```

Comment:
Maybe here mention as well the `ELIXIR_ID=<LS_AAI_CLIENT_ID>` and the `ELIXIR_SECRET=<LS_AAI_CLIENT_SECRET>` and how you obtain them? In general to move this part of REMS section here:

> The values for `oidc-client-id` and `oidc-client-secret` can be obtained from `spreg`, in the Swedish case from `https://services.aai.lifescience-ri.eu/spreg/auth/facilities/detail/<YOUR_PROJECT_ID>`